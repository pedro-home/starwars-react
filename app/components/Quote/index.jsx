import React from 'react';
import PropTypes from 'prop-types';

const Component = ({ text }) => {
  return (
    <h4 className="quote">
      <q>{text}</q>
    </h4>
  );
};

Component.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Component;
