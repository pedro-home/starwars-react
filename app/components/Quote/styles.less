.quote {
  color: #00d6d6;
  text-align: center;
  font-style: italic;
}
