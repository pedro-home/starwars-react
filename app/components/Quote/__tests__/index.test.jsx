import React from 'react';
import { render, getByText } from '@testing-library/react';

import Quote from '../index';

describe('Given <Quote />', () => {
  let component;
  let defaultProps;

  beforeAll(() => {
    defaultProps = { text: 'testing' };
  });

  describe('When initializes', () => {
    beforeEach(() => {
      const { container } = render(<Quote {...defaultProps} />);
      component = container;
    });

    it('Then should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('And should render a text', () => {
      expect(getByText(component, defaultProps.text)).toBeTruthy();
    });
  });
});
