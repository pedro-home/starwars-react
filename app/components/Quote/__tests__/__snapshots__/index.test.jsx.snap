// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Quote /> When initializes Then should match the snapshot 1`] = `
<div>
  <h4
    class="quote"
  >
    <q>
      testing
    </q>
  </h4>
</div>
`;
